src/APICore.ts

```diff
@@ -53,6 +53,14 @@ export default class API {
         return await this.request<T>(url, args);
     }
 
+    async postJSON<T = Record<string, unknown>>(
+        url: string,
+        body: any = {},
+        args: RequestInit = {method: 'post', body: body, headers: {'Content-Type': 'application/json'}}
```

Comment:
Not sure this is valid.
![image](https://user-images.githubusercontent.com/35579930/92614532-28ac2380-f28a-11ea-92eb-53cc69ad17c9.png)
Source: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch

Can't you use the existing `post` method? I'm curious to know in what use case a stringified JSON is not working.

Reply:
I think it's probably the same use case as in ML Advanced configuration. Maybe we could try to pass the `args` in the `put` and `post` methods. Here is the example: https://github.com/coveo/platform-client/blob/master/src/resources/MachineLearning/ModelConfiguration/ModelConfiguration.ts#L20

Reply:
In your example, `modelConfigFileContents` is a string thought, so it works. Here the body seems to be directly an object which is not an appropriate type for `body` according to MDN

Reply:
Ah yes, you're right.